src/oaklib/implementations/gilda.py

```diff
@@ -0,0 +1,39 @@
+"""A text annotator based on Gilda."""
+
+from dataclasses import dataclass
+from typing import Iterator
+
+from oaklib.datamodels.text_annotator import TextAnnotation, TextAnnotationConfiguration
+from oaklib.interfaces import TextAnnotatorInterface
+from oaklib.interfaces.text_annotator_interface import TEXT, nen_annotation
+
+__all__ = [
+    "GildaImplementation",
+]
+
+
+@dataclass
+class GildaImplementation(TextAnnotatorInterface):
+    """Perform named entity normalization on text strings with Gilda [gyori2021]_.
+
+    .. [gyori2021] Benjamin M Gyori, Charles Tapley Hoyt, Albert Steppi (2021)
+        `Gilda: biomedical entity text normalization with machine-learned
+        disambiguation as a service <https://doi.org/10.1093/bioadv/vbac034>`_,
+        *Bioinformatics Advances*, Volume 2, Issue 1, 2022, vbac034,
+    """
+
+    def annotate_text(
+        self, text: TEXT, configuration: TextAnnotationConfiguration = None
+    ) -> Iterator[TextAnnotation]:
+        if not configuration:
+            raise NotImplementedError
+        if not configuration.matches_whole_text:
+            raise NotImplementedError
+
+        import gilda
+
+        matches = gilda.ground(text)
```

Comment:
any thoughts on how to incorporate the context?

(can be done on a separate PR)

I can see two ways

1. a separate context argument in the interface (that many implementations can choose to ignore)
2. treat the full text span as the context, and instead add an optional start/end to the call to the interface, essentially allowing the caller to say "annotate this region of the text using the whole text as context"

2 is maybe a bit more fiddly for the client

Reply:
Gilda supports the context argument out of the box but getting into that calls bigger questions about doing NER vs NEN with the same Interface

Reply:
I’d say just passing it as an extra argument would me much more simple than fiddling around with text positions

Reply:
On second look, this discussion isn't specific to Gilda so maybe it should be considered in a separate issue